File: infrastructure/spring-boot-webflux-rest-api-adapter/src/main/java/codes/fdk/blueprint/api/infrastructure/rest/webflux/PostCategoryRequest.java
package codes.fdk.blueprint.api.infrastructure.rest.webflux;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Pattern;

record PostCategoryRequest(@JsonProperty("name") @NotBlank String name,
                           @JsonProperty("slug") @Pattern(regexp = "[^\s]+") @NotBlank String slug,
                           @JsonProperty("isVisible") boolean isVisible) {}

File: infrastructure/vertx-web-openapi-adapter/src/main/kotlin/codes/fdk/blueprint/api/infrastructure/vertx/web/openapi/PostCategoryRequest.kt
package codes.fdk.blueprint.api.infrastructure.vertx.web.openapi

import com.fasterxml.jackson.annotation.JsonProperty

internal data class PostCategoryRequest(
    @JsonProperty("name") val name: String,
    @JsonProperty("slug") val slug: String,
    @JsonProperty("visible") val visible: Boolean
)
